assignments/14BuildDeployStreamlitDashboard.md

```diff
@@ -0,0 +1,86 @@
+# CTD_Assignment_14: Build and Deploy a Streamlit Dashboard
+
+This assignment is to be implemented using **Streamlit**.  
+You will **import a dataset**, **build a dashboard**, **visualize insights**, **showcase data cleaning**, and **deploy your app** to **Streamlit Community Cloud**.
+
+---
+
+## **Task 1: Import a Dataset**
+- Choose a public dataset (CSV, Excel, or through an API).
+- Import the dataset into your Python script using Pandas.
+- Display the first few rows (`.head()`) to give an overview of the data.
+
+**Tips**: You can find datasets on Kaggle, UCI Machine Learning Repository, or other open sources.
+
+---
+
+## **Task 2: Data Cleaning**
+- Identify any missing or inconsistent values.
+- Handle missing data by either:
+  - Filling (e.g., with mean/median/mode) or
+  - Dropping missing values.
+- Rename columns if necessary for clarity.
+- Display a brief before/after summary of your cleaning process in your app.
+
+**Tip**: Use Streamlit components like `st.write()` and `st.dataframe()` to show the changes.
+
```

Comment:
There is something a little clumsy here: Whenever a streamlit control changes, the whole script runs again, including the cleaning.  That doesn't seem right.  Maybe what should happen is that when the script runs, the dataframe is loaded and displayed, in dirty form.  A button on the page might cause cleaning to occur, so that then the clean dataframe is displayed.  I am not sure how to do this.